---
title: C6320 |Microsoft 文件
ms.custom: ''
ms.date: 11/04/2016
ms.reviewer: ''
ms.suite: ''
ms.technology:
- vs-ide-code-analysis
ms.tgt_pltfrm: ''
ms.topic: article
f1_keywords:
- C6320
helpviewer_keywords:
- C6320
ms.assetid: fb9e568e-b3d4-4ce2-a276-a64ad74d7b1e
caps.latest.revision: 10
author: mikeblome
ms.author: mblome
manager: ghogen
ms.workload:
- multiple
ms.openlocfilehash: dbf498f943fb4f6e446cd254e7ca5f844ee0ec68
ms.sourcegitcommit: 32f1a690fc445f9586d53698fc82c7debd784eeb
ms.translationtype: MT
ms.contentlocale: zh-TW
ms.lasthandoff: 12/22/2017
---
# <a name="c6320"></a>C6320
警告 C6320： 例外狀況篩選條件運算式是常數 EXCEPTION_EXECUTE_HANDLER。 這可能會遮罩不要用來處理的例外狀況  
  
 這個警告表示使用 __except 區塊中的 EXCEPTION_EXECUTE_HANDLER 常數的副作用。 在此情況下中的陳述式\__except 區塊一律會執行處理的例外狀況，包括您不想在特定的函式中處理的例外狀況。 建議您檢查之前處理的例外狀況。  
  
## <a name="example"></a>範例  
 下列程式碼會產生這個警告，因為是 __except 區塊會嘗試處理所有類型的例外狀況：  
  
```  
#include <stdio.h>   
#include <excpt.h>   
  
void f(int *p)   
{   
   __try  
   {   
      puts("in try");   
      *p = 13;  // might cause access violation exception  
      // code ...  
   }   
   __except(EXCEPTION_EXECUTE_HANDLER) // warning  
   {   
      puts("in except");   
      // code ...  
   }   
}   
```  
  
 若要修正這個警告，請使用`GetExceptionCode`來檢查特定的例外狀況，再處理它，如下列程式碼所示：  
  
```  
#include <stdio.h>   
#include <windows.h>   
#include <excpt.h>   
  
void f(int *p)   
{   
   __try  
   {   
      puts("in try");   
      *p = 13;    // might cause access violation exception   
      // code ...  
   }   
   __except(GetExceptionCode()==EXCEPTION_ACCESS_VIOLATION ?   
               EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH)  
   {   
      puts("in except");   
      // code ...  
   }   
}  
```  
  
## <a name="see-also"></a>請參閱  
 [try-except 陳述式](/cpp/cpp/try-except-statement)